### PR TITLE
Add graphviz into Dockerfile images for Python API documentation

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml requires graphviz, which is installed during Docker image creation

*IMPORTANT* numpy version was upgraded from 1.16 to 1.19 in this PR. 

*NOT IMPORTANT FOR RELEASE 1.8*

### CODE REVIEW

**Review by snnn — DISMISSED** [review]

File: tools/ci_build/github/linux/docker/scripts/install_os_deps.sh [hunk 1]
```diff
@@ -19,7 +19,11 @@ done
 
 DEVICE_TYPE=${DEVICE_TYPE:=Normal}
 
-#Download a file from internet
+# Development tools and libraries
+apt install -y \
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/scripts/install_os_deps.sh:23]: You may move it to install_ubuntu.sh

Comment (snnn) [reply tools/ci_build/github/linux/docker/scripts/install_os_deps.sh:23]: This file is shared by centos and ubuntu. But I feel centos images should use the one in the manylinux folder.

Comment (thiagocrepaldi) [reply tools/ci_build/github/linux/docker/scripts/install_os_deps.sh]: just removed it

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 2]
```diff
@@ -132,7 +132,7 @@ stages:
               -e NIGHTLY_BUILD \
               -e BUILD_BUILDNUMBER \
               onnxruntimecpubuild \
-                bash /onnxruntime_src/tools/doc/builddoc.sh $(PythonManylinuxDir)/bin/ /onnxruntime_src /build Release
+                bash -c " $(PythonManylinuxDir)/bin/python3 -m pip install /build/Release/dist/onnxruntime*.whl ; /onnxruntime_src/tools/doc/builddoc.sh $(PythonManylinuxDir)/bin/ /onnxruntime_src /build Release " ;
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:135]: I feel "&&" is better than ";"

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 3]
```diff
@@ -436,13 +436,22 @@ stages:
...
             mkdir -p $HOME/.onnx
             docker run --rm \
+              --device=/dev/kfd \
+              --device=/dev/dri \
+              --group-add $(video) \
+              --group-add $(render) \
+              --privileged \
+              --ipc=host \
+              --network=host \
+              --cap-add=SYS_PTRACE \
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:446]: You already have line 443. Do you still need line 446?

File: tools/ci_build/github/linux/docker/scripts/training/requirements.txt [hunk 4]
```diff
@@ -2,10 +2,10 @@
 -f https://download.pytorch.org/whl/torch_stable.html
 # transformers requires sklearn
 sklearn
-numpy==1.16.6
+numpy
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/scripts/training/requirements.txt:5]: Why this line needs be changed?

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml (hunk 3 above)
Comment (thiagocrepaldi) [reply tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:446]: I am not familiar about ROCM image. I just copied and pasted from previous calls to docker run in this file

File: tools/ci_build/github/linux/docker/scripts/training/requirements.txt (hunk 4 above)
Comment (thiagocrepaldi) [reply tools/ci_build/github/linux/docker/scripts/training/requirements.txt:5]: There are conflicts some conflicts between different numpy and tensorboard in different requirements.txt files. As the packaging pipeline runs 1) build, 2) test, 3) packaging and 4) documenting, all 4 requirements are needed to work together. numpy 1.16.6 is not compatible with all dependencies

Comment (snnn) [reply tools/ci_build/github/linux/docker/scripts/training/requirements.txt:5]: Last night I found training python packaging pipelines(including ROCM) uses a different numpy version than expected. If your code was built with numpy version A, then at runtime it requires at least number version >=A. From the requirements.txt in onnx runtime's top source dir, you can see we claim it requires numpy 1.16.6. I tried to increase it, but the change got rejected. 

I have reported this problem to Liqun and Suffian. They said it is ok for now but eventually it needs be fixed.

Comment (snnn) [reply tools/ci_build/github/linux/docker/scripts/training/requirements.txt:5]: You may resolve this conversation after reading this.

Comment (thiagocrepaldi) [reply tools/ci_build/github/linux/docker/scripts/training/requirements.txt:5]: I think we should not change this to release 1.8. It adds unnecessary risk. Maybe that was the context for the rejection?

I wouldnt recommend cherry-picking this PR to 1.8, but for the next release only

Comment (baijumeswani) [reply tools/ci_build/github/linux/docker/scripts/training/requirements.txt:5]: Do we not need to pin the version number?

**Review by baijumeswani — APPROVED** [review]

File: tools/ci_build/github/linux/docker/scripts/training/requirements.txt (hunk 4 above)
Comment (thiagocrepaldi) [reply tools/ci_build/github/linux/docker/scripts/training/requirements.txt]: I will try to pin it to 1.19.5

**Review by baijumeswani — APPROVED** [review]

**Review by snnn — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 5]
```diff
@@ -132,7 +132,7 @@ stages:
               -e NIGHTLY_BUILD \
               -e BUILD_BUILDNUMBER \
               onnxruntimecpubuild \
-                bash /onnxruntime_src/tools/doc/builddoc.sh $(PythonManylinuxDir)/bin/ /onnxruntime_src /build Release
+                bash -c " $(PythonManylinuxDir)/bin/python3 -m pip install /build/Release/dist/onnxruntime*.whl && /onnxruntime_src/tools/doc/builddoc.sh $(PythonManylinuxDir)/bin/ /onnxruntime_src /build Release " ;
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:135]: If it is a nightly build, then the package name is ort_nightly or ort_gpu_nightly. onnxruntime*.whl doesn't match any.

**Pull request merged**